src/test/java/net/openhft/chronicle/network/internal/SocketExceptionUtilTest.java

```diff
@@ -0,0 +1,91 @@
+package net.openhft.chronicle.network.internal;
+
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.network.TCPRegistry;
+import net.openhft.chronicle.network.tcp.ChronicleServerSocketChannel;
+import net.openhft.chronicle.network.tcp.ChronicleSocketChannel;
+import net.openhft.chronicle.network.tcp.ChronicleSocketChannelFactory;
+import net.openhft.chronicle.threads.Threads;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.Socket;
+import java.net.SocketException;
+import java.nio.ByteBuffer;
+import java.util.Locale;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static net.openhft.chronicle.network.internal.SocketExceptionUtil.isAConnectionResetException;
+import static org.junit.Assert.*;
+
+public class SocketExceptionUtilTest {
+
+    /**
+     * Original means of detection
+     */
+    @Test
+    public void isAConnectionResetExceptionReturnsTrueWhenMessageMatches() {
+        assertTrue(isAConnectionResetException(new IOException("Connection reset by peer")));
+    }
+
+    /**
+     * Added by this change https://github.com/openjdk/jdk/commit/3a4d5db248d74020b7448b64c9f0fc072fc80470
+     * <p>
+     * Thrown in JDK 13 and above
+     */
+    @Test
+    public void isAConnectionResetExceptionReturnsTrueForSocketExceptionWithShorterMessage() {
+        assertTrue(isAConnectionResetException(new SocketException("Connection reset")));
+    }
+
+    @Test
+    public void isAConnectionResetExceptionReturnsFalseForOtherExceptions() {
+        assertFalse(isAConnectionResetException(new SocketException("Something else happened")));
+        assertFalse(isAConnectionResetException(new IOException("Something else happened")));
+    }
+
+    @Test
+    public void isAConnectionResetIsRobustAgainstNullMessages() {
+        assertFalse(isAConnectionResetException(new IOException()));
+    }
+
+    @Test
+    public void testConnectionResetDetectionForLocales() throws IOException {
+        final Locale originalDefault = Locale.getDefault();
+        try {
+            Locale.setDefault(Locale.KOREA);
```

Comment:
Unfortunately, the error messages of the JVM seem to be set in stone on JVM start-up and are not affected by Locale.setDefault and even by `-Duser.language` - they're taken from `$LANG` in the case of Linux, not sure about Windows. So if run with LANG=ja_JP the test will fail every time.

Reply:
I can't seem to reproduce that in 17 or 1.8 (on Linux)
17:
```
09:23:53 (improve_connection_reset_detection) ~/development/projects/Chronicle-Network$ export LANG=ja_JP
09:24:23 (improve_connection_reset_detection) ~/development/projects/Chronicle-Network$ java --version
openjdk 17.0.1 2021-10-19
OpenJDK Runtime Environment (build 17.0.1+12)
OpenJDK 64-Bit Server VM (build 17.0.1+12, mixed mode)
09:24:36 (improve_connection_reset_detection) ~/development/projects/Chronicle-Network$ mvn test -Dtest=SocketExceptionUtilTest -Pjava17
...
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running net.openhft.chronicle.network.internal.SocketExceptionUtilTest
[main] INFO net.openhft.chronicle.core.Jvm - Chronicle core loaded from file:/home/nick/.m2/repository/net/openhft/chronicle-core/2.22ea30/chronicle-core-2.22ea30.jar
[main] WARN net.openhft.chronicle.core.Jvm - Resource tracing is turned on. If you are performance testing or running in PROD you probably don't want this
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.393 s - in net.openhft.chronicle.network.internal.SocketExceptionUtilTest
```

1.8
```
09:27:08 (improve_connection_reset_detection) ~/development/projects/Chronicle-Network$ export LANG=ja_JP
09:27:11 (improve_connection_reset_detection) ~/development/projects/Chronicle-Network$ java -version
openjdk version "1.8.0_312"
OpenJDK Runtime Environment (build 1.8.0_312-b10)
OpenJDK 64-Bit Server VM (build 25.312-b10, mixed mode)
09:27:16 (improve_connection_reset_detection) ~/development/projects/Chronicle-Network$ mvn test -Dtest=SocketExceptionUtilTest
...
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running net.openhft.chronicle.network.internal.SocketExceptionUtilTest
[main] INFO net.openhft.chronicle.core.Jvm - Chronicle core loaded from file:/home/nick/.m2/repository/net/openhft/chronicle-core/2.22ea30/chronicle-core-2.22ea30.jar
[main] WARN net.openhft.chronicle.core.Jvm - Resource tracing is turned on. If you are performance testing or running in PROD you probably don't want this
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.42 s - in net.openhft.chronicle.network.internal.SocketExceptionUtilTest
```

In the case of the Java 17 exception, the message appears to be a static string (see [here](
https://github.com/openjdk/jdk/commit/3a4d5db248d74020b7448b64c9f0fc072fc80470#diff-61a7dbe71a2801bc953733ca886e4420ce2d12a0e816feb71966da6d75e24e13R343)), I'm less sure about the IOException in 1.8 because it's thrown from native code. Perhaps this is the one causing the issue? But I can't seem to reproduce it with openjdk on Linux?

Reply:
Also tried with `ja-JP` (as listed here https://www.oracle.com/java/technologies/javase/jdk8-jre8-suported-locales.html) and no luck

Reply:
Just gonna merge this for now as it will have the desired effect of making out tests less flaky due to the warnings. I am interested in how this localisation works though, it's always seemed a bit of a black art and applied inconsistently to me.

Reply:
You need to have `language-pack-ja` in your system (Ubuntu package; may be different on your system) in order to actually get those messages. The locale (LANG variable) should be `ja_JP.UTF-8` - then it would trigger.

Come to think of it, we can recommend the removal of language pack to customers who stumble on the issue.